x-pack/solutions/security/plugins/security_solution/public/flyout/document_details/right/components/table_tab_tour.tsx

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useCallback, useState, useEffect } from 'react';
+import { i18n } from '@kbn/i18n';
+import type { EuiTourStepProps } from '@elastic/eui';
+import { EuiTourStep, EuiButton, EuiButtonEmpty, EuiText } from '@elastic/eui';
+import { FormattedMessage } from '@kbn/i18n-react';
+import { useKibana } from '../../../../common/lib/kibana';
+import { FLYOUT_STORAGE_KEYS } from '../../shared/constants/local_storage';
+import {
+  FLYOUT_TABLE_PIN_ACTION_TEST_ID,
+  TABLE_TAB_SETTING_BUTTON_TEST_ID,
+  TABLE_TAB_TOUR_TEST_ID,
+} from './test_ids';
+
+const TOUR_TITLE = i18n.translate('xpack.securitySolution.flyout.tour.tableTabPinning.title', {
+  defaultMessage: 'Customize the Table tab',
+});
+
+export const TOUR_STEPS = [
+  {
+    step: 1,
+    title: i18n.translate('xpack.securitySolution.flyout.tour.tableTabPinning.title', {
+      defaultMessage: 'Pin fields',
+    }),
+    content: (
+      <EuiText>
+        <FormattedMessage
+          id="xpack.securitySolution.flyout.tour.tableTabPinning.description"
+          defaultMessage="Pin fields that you're interested in or want to save for later."
+        />
+      </EuiText>
+    ),
+    anchor: `[data-test-subj=${FLYOUT_TABLE_PIN_ACTION_TEST_ID}]`,
+    anchorPosition: 'leftCenter' as EuiTourStepProps['anchorPosition'],
+  },
+  {
+    step: 2,
+    title: i18n.translate('xpack.securitySolution.flyout.tour.tableTabSetting.title', {
+      defaultMessage: 'Display and hide fields',
+    }),
+    content: (
+      <EuiText>
+        <FormattedMessage
+          id="xpack.securitySolution.flyout.tour.tableTabSetting.description"
+          defaultMessage="Modify the tab's settings to display or hide certain fields. For example, choose to only display highlighted fields."
+        />
+      </EuiText>
+    ),
+    anchor: `[data-test-subj=${TABLE_TAB_SETTING_BUTTON_TEST_ID}]`,
+    anchorPosition: 'upCenter' as EuiTourStepProps['anchorPosition'],
+  },
+];
+
+const TOUR_INITIAL_STATE = {
+  currentTourStep: 1,
+  isTourActive: true,
+  tourPopoverWidth: 360,
+  tourSubtitle: 'Customize the Table tab',
+};
+
+/**
+ * A short tour to guide the user through pinning and field settings on the Table tab.
+ */
+export const TableTabTour = ({
+  setIsPopoverOpen,
+}: {
+  setIsPopoverOpen: (isPopoverOpen: boolean) => void;
+}) => {
+  const {
+    services: { storage },
+  } = useKibana();
+
+  const [state, setState] = useState(() => {
+    const initialState = storage.get(FLYOUT_STORAGE_KEYS.TABLE_TAB_TOUR);
+    if (initialState) {
+      return initialState;
+    }
+    return TOUR_INITIAL_STATE;
```

Comment:
nit but this could be simplified to return `initialState || TOUR_INITIAL_STATE`?